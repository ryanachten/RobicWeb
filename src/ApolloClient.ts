import ApolloClient from "apollo-boost";
import { API_URL } from "./constants/API";

const client = new ApolloClient({
  uri: API_URL,
  request: async operation => {
    const token = window.localStorage.getItem("token");
    operation.setContext({
      headers: {
        authorization: token
      }
    });
  }
});

export default client;
