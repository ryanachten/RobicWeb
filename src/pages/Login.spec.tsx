import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import NakedLogin from "./Login";
import renderer from "react-test-renderer";
import { ApolloProvider } from "react-apollo";
import ApolloClient from "../ApolloClient";

describe("Page: Login", () => {
  let shallow: any;

  const initProps = () => {
    return {
      mutate: null,
      match: null,
      history: null,
      location: null
    };
  };

  const Login = () => (
    <ApolloProvider client={ApolloClient}>
      <NakedLogin {...initProps} />
    </ApolloProvider>
  );

  beforeEach(() => {
    shallow = createShallow({ untilSelector: "Login" });
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call update fields properly", () => {
    const wrapper = shallow(<Login />);
    const instance = wrapper.instance();
    expect(instance.state.email).toBe("");
    const input = "testing";
    instance.onFieldUpdate("email", input);
    expect(instance.state.email).toBe(input);
  });
});
