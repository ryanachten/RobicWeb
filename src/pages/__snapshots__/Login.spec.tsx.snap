// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Login renders correctly 1`] = `
<div
  className="Login-root-5"
>
  <form
    onSubmit={[Function]}
  >
    <h1
      className="MuiTypography-root-7 MuiTypography-h1-19 Login-header-3"
    >
      Login
    </h1>
    <div
      className="MuiFormControl-root-43 Login-input-4"
    >
      <label
        className="MuiFormLabel-root-58 MuiInputLabel-root-47 MuiInputLabel-formControl-52 MuiInputLabel-animated-55"
        data-shrink={false}
      >
        Email
      </label>
      <div
        className="MuiInputBase-root-78 MuiInput-root-65 MuiInput-underline-69 MuiInputBase-formControl-79 MuiInput-formControl-66"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-88 MuiInput-input-73"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-43 Login-input-4"
    >
      <label
        className="MuiFormLabel-root-58 MuiInputLabel-root-47 MuiInputLabel-formControl-52 MuiInputLabel-animated-55"
        data-shrink={false}
      >
        Password
      </label>
      <div
        className="MuiInputBase-root-78 MuiInput-root-65 MuiInput-underline-69 MuiInputBase-formControl-79 MuiInput-formControl-66"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-88 MuiInput-input-73 MuiInputBase-inputType-91 MuiInput-inputType-76"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="submitWrapper"
    >
      <button
        className="MuiButtonBase-root-121 MuiButton-root-95 MuiButton-text-97 MuiButton-flat-100"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-96"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
    </div>
  </form>
</div>
`;
